vercel/turbo
# feat: add read only remote cache

### Description

Forking the existing PR of #6326 to see if this helps CI along. See that PR for more detail.

### Testing Instructions

Manual testing:
```
[0 olszewski@chriss-mbp] /Users/olszewski/code/vercel/turborepo $ turbo_dev build --filter=docs --remote-cache-read-only --force
• Packages in scope: docs
• Running build in 1 packages
• Remote caching enabled
docs:rss: cache bypass, force executing 73afc3e91d179e7d
docs:schema: cache bypass, force executing 1783946761b14ceb
@turbo/workspaces:build: cache bypass, force executing a31c222598c45c9a
docs:schema:  WARN  Unsupported engine: wanted: {"node":"20.x"} (current: {"node":"v16.18.1","pnpm":"8.9.0"})
@turbo/workspaces:build:
@turbo/workspaces:build: > @turbo/workspaces@1.10.17-canary.6 build /Users/olszewski/code/vercel/turborepo/packages/turbo-workspaces
@turbo/workspaces:build: > tsup
@turbo/workspaces:build:
docs:rss:  WARN  Unsupported engine: wanted: {"node":"20.x"} (current: {"node":"v16.18.1","pnpm":"8.9.0"})
docs:schema:
docs:schema: > docs@1.0.0 schema /Users/olszewski/code/vercel/turborepo/docs
docs:schema: > turbo-types-generate ./public/schema.json
docs:schema:
docs:rss:
docs:rss: > docs@1.0.0 rss /Users/olszewski/code/vercel/turborepo/docs
docs:rss: > node scripts/generate-rss.js
docs:rss:
 WARNING  Remote cache is read-only, skipping upload
...
```
The warning is only printed once. Without the `--remote-cache-read-only` flag the cache is written to as expected.
